zerver/lib/test_classes.py

```diff
@@ -850,3 +856,85 @@ def setUp(self) -> None:
 
     def setUpBeforeMigration(self, apps: StateApps) -> None:
         pass  # nocoverage
+
+class SlapdObject(slapdtest.SlapdObject):
+    # This class will create some temporary directories and files
+    # under TMPDIR. We override it to /tmp, since otherwise
+    # it uses the current working directory.
+    TMPDIR = "/tmp"
+
+    def setup_rundir(self) -> None:
+        """
+        Override to add our own schemas
+        """
+        super().setup_rundir()
+
+        ldap_fixture_dir = os.path.join(
+            os.path.dirname(__file__),
+            "../tests/fixtures/ldap/"
+        )
+        # Create a link to the schema in slapdtest's internal directory:
+        self._ln_schema_files(['user_extension.schema'], ldap_fixture_dir)
+        # Add to the openldap_schema_files list, so that it gets imported:
+        self.openldap_schema_files.append('user_extension.schema')
+
+class SlapdTestCase(ZulipTestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+        This setup code is quite expensive, so it should be run once
+        for the entire class, rather than having it in normal setUp()
+        and running it for every single test here.
+        """
+        super().setUpClass()
+
+        cls.server = SlapdObject()
+        cls.server.suffix = "dc=zulip,dc=com"
+        cls.server.openldap_schema_files = [
+            "core.schema",
+            "cosine.schema",
+            "inetorgperson.schema",
+        ]
```

Comment:
Can you add a comment explaining this list of schemas?  E.g. are they the standard base schemas present in any LDAP server, or just ones used for this particular purpose?

Reply:
N/A anymore, since things are switched to a fakeldap setup